scene/main/window.cpp

```diff
@@ -648,6 +663,19 @@ void Window::_update_window_size() {
 void Window::_update_viewport_size() {
 	//update the viewport part
 
+#ifndef _3D_DISABLED
+	if (use_xr) {
```

Comment:
Not entirely sure why this needs to be added? When XR is enabled we don't really care about the window size as we're overriding this.

Reply:
So, it turns out that this was the fix for the window size weirdness! Switching to `window_get_real_size()` didn't help (and, actually, I think that's not correct, because on X11, at least, that includes the window decorations, which we don't want).

Without this bit of code, the `BlitToScreen`s returned by `MobileVRInterface` don't work right. The `Rect2i()` that gets passed into `viewport_attach_to_screen()` here is what eventually gets passed to `XRInterface::post_draw_viewport()` as the screen rect. So, we need to keep updating that as the window changes size, in order to have the correct screen rect.

Reply:
Hmmm, I don't know enough about the internals of the window system to know it  isn't getting the correct size here. It should be getting the size of the window, not the viewport. You can have situation in normal apps where the viewport only covers a portion of the screen and is only blit to that portion.

It doesn't make sense that just because we have XR mode on, that suddenly breaks.... There is something else going on here..

Reply:
Well, the viewport is holding on to the window size, which is which is what's being passed into `XRInterface::post_draw_viewport()`. This would probably be easier to walk through on a video call where I could share my screen, but if you walk through the code backwards:

1. In [servers/rendering/renderer_viewport.cpp](https://github.com/godotengine/godot/blob/b52305351d417c47ce019529fb511ecfd72c07e4/servers/rendering/renderer_viewport.cpp#L684) it has:
    ```
    Vector<BlitToScreen> blits = xr_interface->post_draw_viewport(vp->render_target, vp->viewport_to_screen_rect);
    ```
2. The `vp->viewport_to_screen_rect` above is set in [RendererViewport::viewport_attach_to_screen()](https://github.com/godotengine/godot/blob/b52305351d417c47ce019529fb511ecfd72c07e4/servers/rendering/renderer_viewport.cpp#L875)
3. And it's `Window::_update_window_size()` which [normally calls `viewport_attach_to_screen()`](https://github.com/godotengine/godot/blob/b52305351d417c47ce019529fb511ecfd72c07e4/scene/main/window.cpp#L754). We don't want any of the other things that `Window::_update_window_size()` does normally, but when in XR mode, we still need the `viewport_attach_to_screen()` calls so that the code up in point nr 1 still works.

There's probably a better way to ensure that `viewport_attach_to_screen()` gets called when the underlying OS window gets resized than the way I did it, but I'm not sure what that would be.